### PR TITLE
Add inheritance file (from {{InterfaceData}}) and its schema

### PR DESCRIPTION
**Pull request by teoli2003**
This PR imports the data from {{InterfaceData}} (that is the InterfaceData.json from mdn/kumascript/macros).

The format is a bit different, and the schema is defined (as well as a simple markdown file explaining the syntax in layman text).

Note that kumascript macros are still using InterfaceData.json, so it means that for the moment, until these macros are updated, we need to maintain both structure (temporary burden).

### CODE REVIEW

File: api/index.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  css: require('./inheritance'),
```
Comment (Elchi3) [inline api/index.js]: css -> inheritance

File: api/inheritance.schema.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "type": "object",
+  "additionalProperties": {
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+      "inh": {
```
Comment (Elchi3) [inline api/inheritance.schema.json]: I would prefer if we rename "inh" to "inherits" and "impl" to "implements"

**Comment by teoli2003** [comment]
@Elchi3 It is ready for a review, I'm done with the first iteration.

File: README.md [hunk 3]
```diff
@@ -22,6 +22,8 @@ There's a top-level directory for each broad area covered: for example, "api",
 "css", "svg". Inside each of these directories is one or more
 JSON files containing the data.
 
+### api
+Contains information about inheritance hierarchy, and about which interfaces implement which mixins.
```
Comment (Elchi3) [inline README.md]:
```
Contains data about Web APIs:
* API inheritance (interface inheritance and mixin implementations)
```

File: api/inheritance.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# Types
```
Comment (Elchi3) [inline api/inheritance.md]: `# Types -> # Inheritance`

File: api/inheritance.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+[data](https://github.com/mdn/data/blob/master/api/inheritance.json) |
+[schema](https://github.com/mdn/data/blob/master/css/inheritance.schema.json)
+
+The inheritance files defines, for each interface, the parent that they inherit properties and methods from, as well as the mixins that they implements.
```
Comment (Elchi3) [inline api/inheritance.md]: Better:
`Interfaces of Web APIs can inherit from other interfaces or implement [mixins](https://developer.mozilla.org/en-US/docs/Glossary/Mixin). For each interface, this data informs about the inherited (parent) interface and the implemented mixin(s).`

File: api/inheritance.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  "inherits": "Node",
+  "implements": [
+    "ParentNode",
+    "LegacyQueryInterface"
+  ]
+}
+```
+
+The 2 properties are both required.
+* `inherits` (a string): the name of the interface it inherits properties and methods from. If "", it means it doesn't inherit from any interface.
```
Comment (Elchi3) [inline api/inheritance.md]: better: ` if "" (empty string)`
I wonder if this is a good idea to indicate no inheritance. Should it be `null` instead?

File: api/inheritance.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+* `implements` (array of strings): the list of mixins it implements the methods and properties. The array can be empty.
```
Comment (Elchi3) [inline api/inheritance.md]: better: "the list of mixins the interface implements."

File: README.md (hunk 3 above)
Comment (teoli2003) [reply README.md]: Changed.

File: api/inheritance.md (hunk 4 above)
Comment (teoli2003) [reply api/inheritance.md]: Fixed

File: api/inheritance.md (hunk 5 above)
Comment (teoli2003) [reply api/inheritance.md]: Fixed.

File: api/inheritance.md (hunk 7 above)
Comment (teoli2003) [reply api/inheritance.md]: Fixed.

File: api/inheritance.md (hunk 6 above)
Comment (teoli2003) [reply api/inheritance.md]: Yes, null is a better idea. I'm preventing "" in the schema too.

**Comment by teoli2003** [comment]
@Elchi3 This is ready for a second pass of review. I'm using null as no value, and prevented the use of "" in the schema itself.

**Comment by Elchi3** [comment]
This needs two more infrastructure additions in order to have it working.
In `index.js:`
```diff
module.exports = {
+  api: require('./api'),
   css: require('./css'),
   l10n: require('./l10n'),
 }
```

In `package.json`:
```diff
   "files": [
+    "api/*.json",
     "css/*.json",
     "l10n/*.json"
   ],
```
With this added, and by using `npm pack` to pack the package locally, and then installing the local package in a test environment, I was able to test that queries like `data.api.inheritance.Window.implements`give you back the correct array of data.

**Comment by teoli2003** [comment]
@Elchi3 I think we are good there :-)
